pkg/server/createExperiment.go

```diff
@@ -1,3 +1,142 @@
 package server
 
-//reads in the values from computeRecommendations and passes them to request to createExperiment kruize
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/stolostron/recommends/pkg/config"
+	"github.com/stolostron/recommends/pkg/helpers"
+	"github.com/stolostron/recommends/pkg/utils"
+	"k8s.io/klog"
+)
+
+//reads in the values from computeRecommendations and prometheus
+//and passes them as request to createExperiment kruize
+
+var create_experiment_url = config.Cfg.KruizeURL + "/createExperiment"
+
+type CreateExperiment struct {
+	Version                string             `json:"version"`
+	ExperimentName         string             `json:"experiment_name"`
+	ClusterName            string             `json:"cluster_name"`
+	PerformanceProfile     string             `json:"performace_profile"`
+	Mode                   string             `json:"mode"`
+	TargetCluster          string             `json:"target_cluster"`
+	KubernetesObjects      []KubernetesObject `json:"kubernetes_objects"`
+	TrialSettings          TrialSettings
+	RecommendationSettings RecommendationSettings
+}
+
+type KubernetesObject struct {
+	Type       string      `json:"type"`
+	Name       string      `json:"name"`
+	Namespace  string      `json:"namespace"`
+	Containers []Container `json:"containers"`
+}
+
+type Container struct {
+	ContainerImage string `json:"container_image_name"`
+	ContainerName  string `json:"container_name"`
+}
+
+type TrialSettings struct {
+	MeasurementDuration string `json:"measurement_duration"`
+}
+
+type RecommendationSettings struct {
+	Threshold string `json:"threshold"`
+}
+
+func LoadValues(clusterID map[string]string, deployments map[string][]string, context context.Context) {
+
+	var reqBody CreateExperiment
+	var kubeObj KubernetesObject
+	var conObj Container
+	var containerDataClean []string
+
+	// parse the clusterID
+	for name, id := range clusterID {
+		reqBody.ExperimentName = fmt.Sprint(name + "-" + id)
+		parts := strings.Split(name, "_")
+		reqBody.ClusterName = parts[0]
+		kubeObj.Namespace = parts[1]
+
+	}
+	//parse deployment data
+	for deployment, containerData := range deployments {
+		containerDataClean = helpers.RemoveDuplicate(containerData)
+		kubeObj.Name = deployment
+		for _, contData := range containerDataClean {
+			conObj = Container{ContainerImage: "", ContainerName: contData}
```

Comment:
if there will be multiple items in containerDataClean, should we be calling create experiment for each container?

Reply:
Wondering if createExperiment should be inside the `for _, contData := range containerDataClean {` loop?

Reply:
these containers are for each deployment - we are calling createExperiment for every deployment.

Reply:
thanks @SherinV! we should be calling per deployment which current code isn't doing. Will update.

Reply:
Done.